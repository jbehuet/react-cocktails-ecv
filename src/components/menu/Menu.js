import { useHistory, useLocation } from 'react-router-dom';
import './Menu.css';

export function Menu() {
    const history = useHistory();
    const location = useLocation();
    
    return (
        <nav className="mobile-bottom-nav">
            <div className="item" onClick={() => history.push('/')}>
                <div className={`item-content ${(location.pathname === '/' ||Â location.pathname.indexOf('/cocktail/') === 0 ) && 'active'}`}>
                    <i className="fas fa-cocktail"></i>
                    Cocktail
                </div>
            </div>
            <div className="item" onClick={() => history.push('/favorites')}>
                <div className={`item-content ${location.pathname === '/favorites' && 'active'}`}>
                    <i className="fas fa-list-alt"></i>
                    Favorites
                </div>
            </div>
        </nav>
    )
}