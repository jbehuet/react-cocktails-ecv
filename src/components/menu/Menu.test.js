import React from "react";
import { create } from "react-test-renderer";

import { Menu } from "./Menu";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/"
  })
}));

describe("Menu component", () => {
  it("renders correctly", () => {
    const component = create(<Menu />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
