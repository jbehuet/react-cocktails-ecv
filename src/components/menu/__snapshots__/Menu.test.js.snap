// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly 1`] = `
<nav
  className="mobile-bottom-nav"
>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="item-content false"
    >
      <i
        className="fas fa-cocktail"
      />
      Cocktail
    </div>
  </div>
  <div
    className="item"
    onClick={[Function]}
  >
    <div
      className="item-content false"
    >
      <i
        className="fas fa-list-alt"
      />
      Favorites
    </div>
  </div>
</nav>
`;
